### PR TITLE
Moves the cloning of a quote to the helper class

### PR DESCRIPTION
**Pull request by mccottry**
https://app.asana.com/0/544708310157130/710610548302228

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Helper/Data.php [hunk 1]
```diff
@@ -181,6 +181,102 @@ public function getConnectJsUrl()
...
+    /**
+     * Creates a copy of a quote including items, addresses, customer details,
+     * and shipping and tax options
+     *
+     * @param Mage_Sales_Model_Quote $sourceQuote   The quote to be cloned
+     * @param bool $is_for_multipage    Determines if the quote is for payment only, (i.e. with address and shipping data), or multi-page, (i.e. without address and shipping data)
+     *
+     * @return Mage_Sales_Model_Quote  The cloned copy of the source quote
+     */
+    public function cloneQuote(Mage_Sales_Model_Quote $sourceQuote, $is_for_multipage = false ) {
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Data.php]: nit: camelCase for variable (isMultipage)

File: app/code/community/Bolt/Boltpay/Helper/Data.php [hunk 2]
```diff
@@ -181,6 +181,102 @@ public function getConnectJsUrl()
...
+            $clonedQuote
+                ->setBillingAddress($sourceQuote->getBillingAddress())
+                ->setShippingAddress($sourceQuote->getShippingAddress())
+                ->getShippingAddress()
+                ->setShippingMethod($sourceQuote->getShippingAddress()->getShippingMethod())
+                ->save();
+        }
+
+        /*
+         *  Attempting to reset some of the values already set by merge affects the totals passed to
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Data.php]: nit: use inline comment // not block comment /**/

File: app/code/community/Bolt/Boltpay/Helper/Data.php [hunk 3]
```diff
@@ -181,6 +181,102 @@ public function getConnectJsUrl()
             Bolt_Boltpay_Block_Checkout_Boltpay::JS_URL_PROD . "/connect.js";
     }
 
+    /**
+     * Creates a copy of a quote including items, addresses, customer details,
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Data.php]: Let's unify language. "Crates a **clone** of a quote.."

File: app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php [hunk 4]
```diff
@@ -129,10 +129,6 @@ public function getCartDataJs($multipage = true)
             $hintData = $this->getAddressHints($customerSession, $sessionQuote);
             ///////////////////////////////////////////////////////////////
 
-            $orderCreationResponse = '';
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php]: Actually I prefer to keep these variable definitions (as opposed to define this inside nested code below and use outside of the nest level)

Comment (mccottry) [reply app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php]: I have moved this before the try block below.

**Pull request merged**